### ISSUE TITLE
Tensorflow converted model crash on broadcast.

### ISSUE THREAD

**Issue by dimitryn** [XiaoMi/mace#257]
## System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Android
- **NDK version(e.g., 15c)**:
17c
- **GCC version(if compiling for host, e.g., 5.4.0)**:
- **MACE version (Use the command: git describe --long --tags)**:
master
- **Python version(2.7)**: 
2.7
- **Bazel version (e.g., 0.13.0)**:
15

Link to model and yml file : https://drive.google.com/open?id=1ScJDXUuEA3-MYUJzE52OIjXAQsloveRQ

Tensorflow converted model crash in runtime.

```
11-12 07:53:41.601 25092 25092 F DEBUG   : signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
11-12 07:53:41.603 25092 25092 F DEBUG   : Abort message: 'eltwise.cc:852 Check failed: (input0->dim_size() == 4) && ((input1->dim_size() == 0) || (input1->dim_size() == 4 && input1->dim(1) == input0->dim(1) && (input1->dim(0) == input0->dim(0) || input1->dim(0) == 1)) || (input1->dim_size() == 1 && input1->dim(0) == input0->dim(1))) only support broadcast channel dimension'
11-12 07:53:41.603 25092 25092 F DEBUG   :     x0   0000000000000000  x1   00000000000061bd  x2   0000000000000006  x3   0000000000000008
11-12 07:53:41.603 25092 25092 F DEBUG   :     x4   0000000000000000  x5   0000000000000000  x6   0000000000000000  x7   0000000000800000
11-12 07:53:41.603 25092 25092 F DEBUG   :     x8   0000000000000083  x9   0000000010000000  x10  0000007fc2598930  x11  0000000000000001
11-12 07:53:41.603 25092 25092 F DEBUG   :     x12  0000000000000002  x13  0000000000000002  x14  ffffffffffffffff  x15  00205684dfa81898
11-12 07:53:41.603 25092 25092 F DEBUG   :     x16  0000006010879fa8  x17  00000070225ea52c  x18  0000000000000000  x19  00000000000061bd
11-12 07:53:41.603 25092 25092 F DEBUG   :     x20  00000000000061bd  x21  0000000000000083  x22  0000006f8776a6f0  x23  000000000000000a
11-12 07:53:41.603 25092 25092 F DEBUG   :     x24  0000007fc2598990  x25  0000007fc2598990  x26  000000702645ea40  x27  0000006f86dd7640
11-12 07:53:41.603 25092 25092 F DEBUG   :     x28  0000006f86dd77e8  x29  0000007fc2598970  x30  000000702259f760
11-12 07:53:41.603 25092 25092 F DEBUG   :     sp   0000007fc2598930  pc   000000702259f788  pstate 0000000060000000
11-12 07:53:41.607 25092 25092 F DEBUG   :
11-12 07:53:41.607 25092 25092 F DEBUG   : backtrace:
11-12 07:53:41.607 25092 25092 F DEBUG   :     #00 pc 000000000001d788  /system/lib64/libc.so (abort+120)
11-12 07:53:41.607 25092 25092 F DEBUG   :     #01 pc 0000000000138fac  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so (mace::logging::LogMessage::GenerateLogMessage()+888)
11-12 07:53:41.607 25092 25092 F DEBUG   :     #02 pc 00000000001390f4  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so (mace::logging::LogMessage::~LogMessage()+88)
11-12 07:53:41.607 25092 25092 F DEBUG   :     #03 pc 0000000000139214  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so (mace::logging::LogMessage::~LogMessage()+24)
11-12 07:53:41.607 25092 25092 F DEBUG   :     #04 pc 00000000000a0a8c  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so (_ZN4mace7kernels9EltwiseOpILNS_10DeviceTypeE0EfE9DoEltwiseIfEENS_10MaceStatusEPKNS_6TensorES8_PS6_+516)
11-12 07:53:41.607 25092 25092 F DEBUG   :     #05 pc 000000000009f5f8  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so
11-12 07:53:41.607 25092 25092 F DEBUG   :     #06 pc 00000000000f2b50  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so
11-12 07:53:41.607 25092 25092 F DEBUG   :     #07 pc 000000000003c1e4  /data/app/com.intel.realsensestudio.faceauthentication-cVv78smA_Wk1UNpkQt6Z_A==/lib/arm64/libmace.so (mace::MaceEngine::Impl::Run(std::__ndk1::map<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>, mace::MaceTensor, std::__ndk1::less<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>>>, std::__ndk1::allocator<std::__ndk1::pair<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std

```

**Comment by yejw5** [comment]
@dimitryn Could you afford the inputs' shapes of eltwise in your model? Currently, eltwise in Mace only supports some specific situation.

**Comment by dimitryn** [comment]
@yejw5  I provided a model , can you tell me which layer you asking to change eltwise ?

**Comment by dimitryn** [comment]
@yejw5 
Sort by execution
Final ops:
input (Transpose): [[1L, 1L, 128L, 128L]]
Resface/Conv1/Conv1_pre/Relu (Conv2D): [[1L, 64L, 64L, 64L]]
Resface/Conv1/Conv_1/Conv_1_1/Conv/Relu (Conv2D): [[1L, 64L, 64L, 64L]]
Resface/Conv1/Conv_1/Conv_1_1/Conv_1/Relu (Conv2D): [[1L, 64L, 64L, 64L]]
Resface/Conv1/Conv_1/Conv_1_1/add (Eltwise): [[1L, 64L, 64L, 64L]]
Resface/Conv1/Conv_1/Conv_1_2/Conv/Relu (Conv2D): [[1L, 64L, 64L, 64L]]
Resface/Conv1/Conv_1/Conv_1_2/Conv_1/Relu (Conv2D): [[1L, 64L, 64L, 64L]]
Resface/Conv1/Conv_1/Conv_1_2/add (Eltwise): [[1L, 64L, 64L, 64L]]
Resface/Conv2/Conv2_pre/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_1/Conv/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_1/Conv_1/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_1/add (Eltwise): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_2/Conv/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_2/Conv_1/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_2/add (Eltwise): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_3/Conv/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_3/Conv_1/Relu (Conv2D): [[1L, 128L, 32L, 32L]]
Resface/Conv2/Conv_2/Conv_2_3/add (Eltwise): [[1L, 128L, 32L, 32L]]
Resface/Conv3/Conv3_pre/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_1/Conv/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_1/Conv_1/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_1/add (Eltwise): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_2/Conv/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_2/Conv_1/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_2/add (Eltwise): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_3/Conv/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_3/Conv_1/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_3/add (Eltwise): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_4/Conv/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_4/Conv_1/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_4/add (Eltwise): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_5/Conv/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_5/Conv_1/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_5/add (Eltwise): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_6/Conv/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_6/Conv_1/Relu (Conv2D): [[1L, 256L, 16L, 16L]]
Resface/Conv3/Conv_3/Conv_3_6/add (Eltwise): [[1L, 256L, 16L, 16L]]
Resface/Conv4/Conv4_pre/Relu (Conv2D): [[1L, 512L, 8L, 8L]]
Resface/Conv4/Conv4/Conv4_1/Conv/Relu (Conv2D): [[1L, 512L, 8L, 8L]]
Resface/Conv4/Conv4/Conv4_1/Conv_1/Relu (Conv2D): [[1L, 512L, 8L, 8L]]
Resface/Conv4/Conv4/Conv4_1/add (Eltwise): [[1L, 512L, 8L, 8L]]
Resface/Logits/Reshape (Reshape): [[1L, 32768L]]
Resface/Bottleneck/MatMul (MatMul): [[1L, 512L]]
Resface/Bottleneck/batch_normalization/batchnorm/add (Eltwise): [[512L]]
Resface/Bottleneck/batch_normalization/batchnorm/Rsqrt (Eltwise): [[512L]]
Resface/Bottleneck/batch_normalization/batchnorm/mul (Eltwise): [[512L]]
Resface/Bottleneck/batch_normalization/batchnorm/mul_1 (Eltwise): [[1L, 512L]]
Resface/Bottleneck/batch_normalization/batchnorm/mul_2 (Eltwise): [[512L]]
Resface/Bottleneck/batch_normalization/batchnorm/sub (Eltwise): [[512L]]
Resface/Bottleneck/batch_normalization/batchnorm/add_1 (Eltwise): [[1L, 512L]]
mace_output_node_output_features (Identity): [[1L, 512L]]
Add quantize tensor range
start optimize memory.
total op: 51
(0, 'CPU_BUFFER', [1048576L])
(1, 'CPU_BUFFER', [1048576L])
(2, 'CPU_BUFFER', [1048576L])

**Comment by dimitryn** [comment]
Hi @yejw5  any update ?

**Comment by yejw5** [comment]
The error message means: the inputs of Eltwise should meet the requirements `(input0->dim_size() == 4) && ((input1->dim_size() == 0) || (input1->dim_size() == 4 && input1->dim(1) == input0->dim(1) && (input1->dim(0) == input0->dim(0) || input1->dim(0) == 1)) || (input1->dim_size() == 1 && input1->dim(0) == input0->dim(1)))`
